crates/ruma-common/src/events/room/message/sanitize.rs

```diff
@@ -0,0 +1,202 @@
+/// Sanitize the given HTML string.
+///
+/// This removes the [tags and attributes] that are not listed in the Matrix specification.
+///
+/// It can also optionally remove the [rich reply fallback].
+///
+/// [tags and attributes]: https://spec.matrix.org/v1.2/client-server-api/#mroommessage-msgtypes
+/// [rich reply fallback]: https://spec.matrix.org/v1.2/client-server-api/#fallbacks-for-rich-replies
+#[cfg(feature = "sanitize")]
+pub fn sanitize_html(s: &str, remove_reply_fallback: RemoveReplyFallback) -> String {
+    let allowed_tags = [
```

Comment:
Is there a possibility for a blacklist instead of a whitelist?

Otherwise, can this list of tags be sourced from the matrix spec or such?

In general, I believe hiding this kind of behavior deep inside the library can become unexpected, as this will do much more than just remove a reply fallback, if some client authors try to do something funky with the HTML.

Because of that, I believe this should be better documented, or the "remove reply fallback" bit moved to a different function that functions on a blacklist, or this list annotated with sources to the spec that this is the intended set of tags that're allowed.

Otherwise, we're essentially walking away from the spec, and introducing some opinionated bits under the client authors' noses, all done automatically when enabling `sanitize`.

Reply:
Also, this should be a `pub const` that exists outside of this function, so that library users can also take a look at what tags are allowed.

---

A thought: Maybe this could be an `impl fn` on a `Sanitizer` struct, which takes those above consts by default, but would allow client authors to introduce additional tags, attrs, and schemas to sanitize for.

We shouldn't restrict this just to this list, but we should give this list (with proper documentation and spec ref) for library users to refer to.

Reply:
I see the tags and such are documented above, apologies, I didn't notice them.

The rest of my comments still stand, I think that could be refactored as a `pub const` for library users to see (with the appropriate doccomments)

Reply:
> Is there a possibility for a blacklist instead of a whitelist?

Sadly no. I hadn't planned to sanitize the HTML in the first place, as this was mostly to remove the reply fallback, and I would have liked to have different methods, but ammonia allows only to whitelist.

Reply:
> Also, this should be a pub const that exists outside of this function, so that library users can also take a look at what tags are allowed.

That's actually a good idea.

> A thought: Maybe this could be an impl fn on a Sanitizer struct, which takes those above consts by default, but would allow client authors to introduce additional tags, attrs, and schemas to sanitize for.

Honestly this method only uses ammonia with the lists, so I think if someone wants to have a customized behavior, they can just implement it themselves. Right now we apply the spec, like we always do.

Reply:
That's a bit annoying. Then i think moving the consts out of the function and making them `pub` would be useful enough, making this function less of a black box.

It'd also maybe help for any library users who'd wanna implement their own sanitization (for whatever reason) and just use a spec-referenced/verified list of tags and attributes.

Reply:
So I mad the lists const but you still have to look at the source to see them...